pkg/lib/machines.js

```diff
@@ -175,23 +175,18 @@
             // wrap values in variants for D-Bus call; at least values.port can
             // be int or string, so stringify everything but the "visible" boolean
             var values_variant = {};
-            for (var prop in values)
+            for (var prop in values) {
                 if (values[prop] !== null) {
                     if (prop == "visible")
                         values_variant[prop] = cockpit.variant('b', values[prop]);
                     else
                         values_variant[prop] = cockpit.variant('s', values[prop].toString());
                 }
+            }
 
             // FIXME: investigate re-using the proxy from Loader (runs in different frame/scope)
             var bridge = cockpit.dbus(null, { bus: "internal", "superuser": "try" });
             var mod = bridge.call("/machines", "cockpit.Machines", "Update", [ "99-webui.json", host, values_variant ])
-                .done(function() {
-                    var prop, over = { };
-                    for (prop in values)
-                        over[prop] = null;
-                    self.overlay(host, over);
-                })
```

Comment:
I remember keeping this code back then as I wasn't sure if `overlay` was still being used for anything except `localhost` (which we don't store on disk). But indeed it looks like `update_session_machine()` is also just called for localhost. I went through all `overlay` in the code and can't find a non-localhost usage any more (and indeed that wouldn't be intended, we always want to poke changes into the d-bus backend which should write it onto the disk).